ambry-frontend/src/main/java/com/github/ambry/frontend/GetAccountsHandler.java

```diff
@@ -121,7 +122,14 @@ private Callback<Void> securityProcessRequestCallback() {
     private Callback<Void> securityPostProcessRequestCallback() {
       return buildCallback(frontendMetrics.getAccountsSecurityPostProcessRequestMetrics, securityCheckResult -> {
         byte[] serialized;
-        if (RestUtils.getRequestPath(restRequest).matchesOperation(ACCOUNTS_CONTAINERS)) {
+        if (RestUtils.getRequestPath(restRequest).matchesOperation(ACCOUNTS_CONTAINERS_DATASETS)) {
+          LOGGER.debug("Received request for getting single dataset with arguments: {}", restRequest.getArgs());
+          Dataset dataset = getDataset();
+          serialized = AccountCollectionSerde.serializeDatasetsInJson(dataset);
+          restResponseChannel.setHeader(RestUtils.Headers.TARGET_ACCOUNT_NAME, dataset.getAccountName());
+          restResponseChannel.setHeader(RestUtils.Headers.TARGET_CONTAINER_NAME, dataset.getContainerName());
+          restResponseChannel.setHeader(RestUtils.Headers.TARGET_DATASET_NAME, dataset.getDatasetName());
```

Comment:
we probably don't have to set the dataset name in the response header since we already have it in the response body.

Reply:
Removed.